packages/vue/test/tracing/tracingMixin.test.ts

```diff
@@ -0,0 +1,238 @@
+  };
+});
+
+vi.mock('../../src/vendor/components', () => {
+  return {
+    formatComponentName: vi.fn().mockImplementation(vm => {
+      return vm.componentName || 'TestComponent';
+    }),
+  };
+});
+
+const mockSpanFactory = (): { name?: string; op?: string; end: Mock; startChild: Mock } => ({
+  name: undefined,
+  op: undefined,
+  end: vi.fn(),
+  startChild: vi.fn(),
+});
+
+vi.useFakeTimers();
+
+describe('Vue Tracing Mixins', () => {
+  let mockVueInstance: any;
+  let mockRootInstance: any;
+
+  beforeEach(() => {
+    vi.clearAllMocks();
+
+    mockRootInstance = {
+      $root: null,
+      componentName: 'RootComponent',
+      $_sentrySpans: {},
+    };
+    mockRootInstance.$root = mockRootInstance; // Self-reference for root
+
+    mockVueInstance = {
+      $root: mockRootInstance,
+      componentName: 'TestComponent',
+      $_sentrySpans: {},
+    };
+
+    (getActiveSpan as any).mockReturnValue({ id: 'parent-span' });
+    (startInactiveSpan as any).mockImplementation(({ name, op }: { name: string; op: string }) => {
+      const newSpan = mockSpanFactory();
+      newSpan.name = name;
+      newSpan.op = op;
+      return newSpan;
+    });
+  });
+
+  afterEach(() => {
+    vi.clearAllTimers();
+  });
+
+  describe('Mixin Creation', () => {
+    it('should create mixins for default hooks', () => {
+      const mixins = createTracingMixins();
+
+      DEFAULT_HOOKS.forEach(hook => {
+        const hookPairs = {
+          mount: ['beforeMount', 'mounted'],
+          update: ['beforeUpdate', 'updated'],
+          destroy: ['beforeDestroy', 'destroyed'],
+          unmount: ['beforeUnmount', 'unmounted'],
+          create: ['beforeCreate', 'created'],
+          activate: ['activated', 'deactivated'],
+        };
+
+        if (hook in hookPairs) {
+          hookPairs[hook as keyof typeof hookPairs].forEach(lifecycleHook => {
+            expect(mixins).toHaveProperty(lifecycleHook);
+            // @ts-expect-error we check the type here
+            expect(typeof mixins[lifecycleHook]).toBe('function');
+          });
+        }
+      });
+    });
+
+    it('should always include the activate and mount hooks', () => {
+      const mixins = createTracingMixins({ hooks: undefined });
+
+      expect(Object.keys(mixins)).toEqual(['activated', 'deactivated', 'beforeMount', 'mounted']);
+    });
+
+    it('should create mixins for custom hooks', () => {
+      const mixins = createTracingMixins({ hooks: ['update'] });
+
+      expect(Object.keys(mixins)).toEqual([
+        'beforeUpdate',
+        'updated',
+        'activated',
+        'deactivated',
+        'beforeMount',
+        'mounted',
+      ]);
+    });
+  });
+
+  describe('Root Component Behavior', () => {
+    it('should always create root span for root component regardless of tracking options', () => {
+      const mixins = createTracingMixins({ trackComponents: false });
+
+      mixins.beforeMount.call(mockRootInstance);
```

Comment:
q (for my understanding mostly): What happens if I call something like `beforeCreate` or another hook coming in prior to the mount hook? the span would end if the `-ed` hook emits, right?